synapse/handlers/account_validity.py

```diff
@@ -81,11 +84,21 @@ async def _send_renewal_emails(self):
                     user_id=user["user_id"], expiration_ts=user["expiration_ts_ms"]
                 )
 
-    async def send_renewal_email_to_user(self, user_id: str):
+    async def send_renewal_email_to_user(self, user_id: str) -> None:
+        """
+        Send a renewal email for a specific user.
+
+        Note that if the user is not set to renew at some point in the future then
+        no email will be sent.
+
+        Args:
+            user_id: The user ID to send a renewal email for.
+        """
         expiration_ts = await self.store.get_expiration_ts_for_user(user_id)
-        await self._send_renewal_email(user_id, expiration_ts)
+        if expiration_ts is not None:
+            await self._send_renewal_email(user_id, expiration_ts)
```

Comment:
Or perhaps this should just raise an error instead of silently ignoring it?

Reply:
Yeahhh, it's probably poor form to return a `200 OK` for a request that will never actually send the email.

At the same time, I could see raising an error potentially breaking someone's script that just loops over their entire userbase.

Perhaps we can raise an error and put out a notice in the release notes just in case?

Reply:
I decided to make the changelog entry into a bugfix about this instead. Hopefully that's enough of a notice?